library/src/test/java/com/getstream/sdk/chat/models/ChannelTest.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    @Test
+    void sendReactionParamsTest() {
+        MessageCallback callback = mock(MessageCallback.class);
+        String messageId = "testMessageId";
+        String reactionType = "like";
+        Map<String, Object> extra = Collections.singletonMap("testKey", "testValue");
+
+        channel.sendReaction(messageId, reactionType, extra, callback);
+
+        verify(client).sendReaction(argThat(argument -> argument.getReaction().getMessageId().equals(messageId)
+                        && argument.getReaction().getType().equals(reactionType)
+                        && argument.getReaction().getExtraData().equals(extra)),
+                eq(callback));
+    }
+
+    @Test
+    void sendReactionTest() {
+        MessageCallback callback = mock(MessageCallback.class);
+        Reaction reaction = new Reaction();
+
+        channel.sendReaction(reaction, callback);
+
+        verify(client).sendReaction(argThat(argument -> argument.getReaction().equals(reaction)), eq(callback));
+    }
+
+    @Test
+    void sendEventTest() {
+        EventCallback callback = mock(EventCallback.class);
+
+        channel.sendEvent(EventType.MESSAGE_READ, callback);
+
+        verify(client).sendEvent(eq(channel),
+                argThat(argument ->
+                        argument.getEvent().get("type").equals(EventType.MESSAGE_READ.label)),
+                eq(callback));
+    }
+
+    @Test
+    void watchTest() {
+        ChannelWatchRequest request = new ChannelWatchRequest();
+        QueryWatchCallback callback = mock(QueryWatchCallback.class);
+
+        channel.watch(request, callback);
+
+        verify(client).queryChannel(eq(channel), eq(request), any(QueryChannelCallback.class));
+    }
+
+    @Test
+    void queryTest() {
+        ChannelWatchRequest request = new ChannelWatchRequest();
+        QueryChannelCallback callback = mock(QueryChannelCallback.class);
+
+        channel.query(request, callback);
+
+        verify(client).queryChannel(channel, request, callback);
+    }
+
+    @Test
+    void getRepliesTest() {
+        GetRepliesCallback callback = mock(GetRepliesCallback.class);
+        String parentId = "testParentId";
+        int limit = 10;
+        String firstMessageId = "testMessageId";
+
+        channel.getReplies(parentId, limit, firstMessageId, callback);
+
+        verify(client).getReplies(parentId, limit, firstMessageId, callback);
+    }
+
+    @Test
+    void updateMessageTest() {
+        MessageCallback callback = mock(MessageCallback.class);
+        Message message = new Message();
+
+        channel.updateMessage(message, callback);
+
+        verify(client).updateMessage(message, callback);
+    }
+
+    @Test
+    void deleteMessageTest() {
+        MessageCallback callback = mock(MessageCallback.class);
+        Message message = new Message();
+
+        channel.deleteMessage(message, callback);
+
+        verify(client).deleteMessage(message.getId(), callback);
```

Comment:
Never like active record. It forces developers to write eventually senseless tests.

Reply:
Actually I think it's a not good approach to have methods, like that, in the model class. It would be perfect to have "kotlin extensions" here :). 
Do you have ideas about how to avoid senseless tests like that?